internal/test/testdata/query_planner/suite/harry.yaml

```diff
@@ -131,3 +132,23 @@ tests:
                         - variable: t
                         - variable: request.resource.attr.teamId
                   - variable: t
+  - action: just-index
+    resource:
+      kind: leave_request
+      policyVersion: default
+    want:
+      kind: KIND_CONDITIONAL
+      condition:
+        expression:
+          operator: in
+          operands:
+            - value: abc123
+            - expression:
+                operator: map
+                operands:
+                  - variable: request.resource.attr.accounts_people
+                  - expression:
+                      operator: lambda
+                      operands:
+                        - variable: ap.account_id
+                        - variable: ap
```

Comment:
I think we at least need to follow some convention so that the "capture variable" always comes first in the list. Otherwise the user has no way of distinguishing which is which. If they follow the above plan and produce `map(ap.account_id, ap)` that'd be wrong.

Reply:
This is a bit tricky but resolution is deterministic. The lambda argument variable is always the second operand. The lambda body is a compound variable, and the lambda argument isn't. 

Compound variables are the results of expression simplification. 
So the alternative is skipping simplification in this case. The result can be:
```yaml
kind: KIND_CONDITIONAL
      condition:
        expression:
          operator: in
          operands:
            - value: "123"
            - expression:
                operator: map
                operands:
                  - variable: request.resource.attr.accounts_people
                  - expression:
                      operator: lambda
                      operands:
                        - expression: ap.account_id
                            operator: get-field
                            operands:
                              - variable: ap      
                              - variable: account_id
                        - variable: ap
```

Again, the order of operands of the `get-field` expression matters.

Reply:
It's tricky. I like this second version better because there's at least a way to  figure out which is which. In the first version, if the convention is that the second operand is always the argument variable, then we should document that because it's not obvious. I would have expected it to be the first operand because that's how `map` expressions are written.

Reply:
`map(ap, ap)` would be a valid expression as well, wouldn't it? So, I think we definitely need to have a documented convention.

Reply:
`x.map(t, t)` is no-op and should be replaced with `x`.